packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -279,28 +279,39 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
         ? 0
         : -1;
 
-    cells.push(
-      <rootProps.components.Cell
-        key={column.field}
-        value={cellParams.value}
-        field={column.field}
-        width={column.computedWidth}
-        rowId={rowId}
-        height={rowHeight}
-        showRightBorder={showRightBorder}
-        formattedValue={cellParams.formattedValue}
-        align={column.align || 'left'}
-        cellMode={cellParams.cellMode}
-        colIndex={indexRelativeToAllColumns}
-        isEditable={cellParams.isEditable}
-        hasFocus={hasFocus}
-        tabIndex={tabIndex}
-        className={clsx(classNames)}
-        {...rootProps.componentsProps?.cell}
-      >
-        {content}
-      </rootProps.components.Cell>,
+    const cellColSpanInfo = apiRef.current.unstable_getCellColSpanInfo(
+      rowId,
+      indexRelativeToAllColumns,
     );
+
+    if (cellColSpanInfo && !cellColSpanInfo.collapsedByColSpan) {
+      const { colSpan, width, other: otherCellProps } = cellColSpanInfo.cellProps;
+
+      cells.push(
+        <rootProps.components.Cell
+          key={column.field}
+          value={cellParams.value}
+          field={column.field}
+          width={width}
+          rowId={rowId}
+          height={rowHeight}
+          showRightBorder={showRightBorder}
+          formattedValue={cellParams.formattedValue}
+          align={column.align || 'left'}
+          cellMode={cellParams.cellMode}
+          colIndex={indexRelativeToAllColumns}
+          isEditable={cellParams.isEditable}
+          hasFocus={hasFocus}
+          tabIndex={tabIndex}
+          className={clsx(classNames)}
+          colSpan={colSpan}
```

Comment:
> This prop is not used inside the cell

I think it may be useful for custom `Cell` component.

> You could use it to set aria-colspan.

It's already set in https://github.com/cherniavskii/material-ui-x/blob/column-spanning/packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnSpanning.ts#L87, but I will move it to `GridRow` to make it more explicit